dubbo-common/src/main/java/org/apache/dubbo/common/extension/support/ActivateComparator.java

```diff
@@ -44,75 +44,85 @@ public int compare(Object o1, Object o2) {
-                || a2Before.length > 0 || a2After.length > 0)
-                && inf != null && inf.isAnnotationPresent(SPI.class)) {
+        if ((a1.before.length > 0 || a1.after.length > 0 || a2.before.length > 0 || a2.after.length > 0) && inf != null) {
             ExtensionLoader<?> extensionLoader = ExtensionLoader.getExtensionLoader(inf);
-            if (a1Before.length > 0 || a1After.length > 0) {
+            if (a1.before.length > 0 || a1.after.length > 0) {
                 String n2 = extensionLoader.getExtensionName(o2.getClass());
-                for (String before : a1Before) {
+                for (String before : a1.before) {
                     if (before.equals(n2)) {
                         return -1;
                     }
                 }
-                for (String after : a1After) {
+            for (String after : a1.after) {
                     if (after.equals(n2)) {
                         return 1;
                     }
                 }
             }
-            if (a2Before.length > 0 || a2After.length > 0) {
+            if (a2.before.length > 0 || a2.after.length > 0) {
                 String n1 = extensionLoader.getExtensionName(o1.getClass());
-                for (String before : a2Before) {
+                for (String before : a2.before) {
                     if (before.equals(n1)) {
                         return 1;
                     }
                 }
-                for (String after : a2After) {
+                for (String after : a2.after) {
                     if (after.equals(n1)) {
                         return -1;
                     }
                 }
             }
         }
-        int n1 = a1 == null ? 0 : a1Order;
-        int n2 = a2 == null ? 0 : a2Order;
+        int n1 = a1 == null ? 0 : a1.order;
+        int n2 = a2 == null ? 0 : a2.order;
         // never return 0 even if n1 equals n2, otherwise, o1 and o2 will override each other in collection like HashSet
         return n1 > n2 ? 1 : -1;
     }
 
+    private Class<?> findSpi(Class clazz) {
+        if (clazz.getInterfaces().length <= 0) {
+            return null;
+        }
+
+        for (Class<?> intf : clazz.getInterfaces()) {
+            if (intf.isAnnotationPresent(SPI.class)) {
+                return intf;
+            } else {
+                Class result = findSpi(intf);
+                if (result != null) {
+                    return result;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    private ActivateInfo parseActivate(Class<?> clazz) {
+        ActivateInfo info = new ActivateInfo();
+        if (clazz.isAnnotationPresent(Activate.class)) {
+            Activate activate = clazz.getAnnotation(Activate.class);
+            info.before = activate.before();
+            info.after = activate.after();
+            info.order = activate.order();
+        } else {
+            com.alibaba.dubbo.common.extension.Activate activate = clazz.getAnnotation(

```

Comment:
Looking at package import **com.alibaba.dubbo.common** is it knowing we have given or just by mistake has come?

Reply:
it's for backward compatibility.